src/ShaderGen/Hlsl/HlslBackend.cs

```diff
@@ -185,6 +185,27 @@ HashSet<ResourceDefinition> resourcesUsed
                 {
                     if (!resourcesUsed.Contains(rd))
                     {
+                        switch (rd.ResourceKind)
```

Comment:
I'd rather we just "revert" the other change and check whether the resource is used inside the other switch statement. This way is a bit worse because now we have two switch statements with the slot incrementing done in two different places.